transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,251 @@
-import csv
-import io
-from typing import Set, Tuple
+from datetime import timedelta
+from typing import Dict, Optional
 
+from transit_odp.common.csv import CSVBuilder, CSVColumn
 from transit_odp.organisation.models import TXCFileAttributes
+from transit_odp.organisation.models.data import SeasonalService, ServiceCodeExemption
 from transit_odp.otc.models import Service as OTCService
+from transit_odp.publish.requires_attention import (
+    get_txc_map,
+    requires_attention,
+    requires_attention_data,
+)
 
+STALENESS_STATUS = [
+    "Stale - End date passed",
+    "Stale - 12 months old",
+    "Stale - OTC Variation",
+]
 
-class ServiceCodesCSV:
-    REGISTERED = "Registered"
-    MISSING_DATA = "Missing data"
 
-    def __init__(self, organisation_id: int):
-        self._organisation_id = organisation_id
-        self.org_service_code_to_dataset = list(
-            TXCFileAttributes.objects.filter(
-                revision__dataset__organisation_id=organisation_id
-            )
-            .get_active_live_revisions()
-            .select_related("revision__dataset")
-            .values_list("service_code", "revision__dataset_id")
+def add_seasonal_service_status(otc_service: dict) -> str:
```

Comment:
You're not adding anything here, you're getting the seasonal service status. Should be called `get_seasonal_service_status`.